File: output/llama_4_maverick/avatar/direct_translation/python/java/atcoder_ABC123_C.java

import java.util.Scanner;

public class atcoder_ABC123_C {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] A = new int[5];
        for (int i = 0; i < 5; i++) {
            A[i] = sc.nextInt();
        }
        int min = A[0];
        for (int i = 1; i < 5; i++) {
            if (A[i] < min) {
                min = A[i];
            }
        }
        System.out.println((int) Math.ceil((double) N / min) + 4);
    }
}

File: output/llama_4_maverick/avatar/direct_translation/java/python/atcoder_ABC123_C.py

import math

def p(ans=None):
    if ans is None:
        print()
    else:
        print(ans)

def main():
    N = int(input())
    min_val = float('inf')
    for _ in range(5):
        val = int(input())
        min_val = min(min_val, val)
    p(math.ceil(N / min_val) + 4)

if __name__ == "__main__":
    main()

